### DISCUSSION TITLE
Lambda Annotations - Use existing ApiGateway

### DISCUSSION THREAD

**Discussion by cacowen** [aws/aws-lambda-dotnet#1752] [Q&A] [answered]
Is it possible to use an existing Api Gateway when creating a lambda function with annotations? I was half expecting to be able to pass in something to the RestApi attribute for ApiGateway much like you can pass in Policies or Role into the LambdaFunction attribute.

When I tried to do this manually I think I ran into an issue with the SAM template. Does SAM support using a pre-existing gateway? If not, can the Lambda Annotations be used with non SAM templates to generate CloudFormation resources?

-- update:
```Seems like I can use CFN (or non SAM template) if I do not set the metadata tool and remove the transform property. This allows me to write full CF without it getting overridden by the annotations. Which allows me to specify an existing ApiGateway for new functions.```

I also thought I saw the underlying generator support both Json and Yaml. How do I get it to use Yaml? Is that a setting in the `aws-lambda-tools-defaults.json` file?

-- update:
```I found out that I can name the template something like `template.yml` and it will be yaml based. although with red squiggly errors (mentioned in a different discussion)``` but now how do i get it from removing comments?

**Comment by bhoradc** [comment, marked as answer]
Hi @cacowen, 

Please feel free to submit a Feature request, as this isn't possible currently. 

Also, I would recommend to open an Issue ([Bug Report](https://github.com/aws/aws-lambda-dotnet/issues/new/choose)) with as much information you can regarding below comments error. 

> I found out that I can name the template something like `template.yml` and it will be yaml based. although with red squiggly errors (mentioned in a different discussion) but now how do i get it from removing comments?

Regards, 
Chaitanya
